src/event-manager.js

```diff
@@ -87,7 +87,10 @@ function handleDelegatedEvent(event) {
       }
     }
 
-    target = target.parentNode;
+    const parent = target.parentNode;
+    const parentIsShadowRoot = parent && parent.toString() === '[object ShadowRoot]';
+
+    target = parentIsShadowRoot ? parent.host : parent;
```

Comment:
@EisenbergEffect For your awareness, this aligns our delegation strategy with the way it works natively with trigger, but could be a potential breaking change. One way to avoid it is to add a flag on event manager, but depends on how we would look at it, since it's an app wide configuration, and one plugin requiring it would enforce this on the rest

Reply:
I agree it could impact existing setups but only if you use shadow dom of course and have elements within shadow dom firing events. Anyone who is using such a setup probably used a workaround like I did.

I currently implemented 2 different workarounds to be able to use delegate on custom components using shadow dom. One workaround is to add an element in the light dom and trigger an event on this element when the shadow dom variant is clicked for example, this acts as a proxy. This scenario might end up producing 2 events now unless you stop the event bubbling than it would still work as it used to. 
The other workaround I used is capture the original event in the shadow dom and fire a custom event on the shadow host. This scenario would still function as expected and should not be impacted.

Reply:
That sounds reasonable to me. Maybe we could just have it this way

Reply:
FYI in my project I just confirmed that implementation 1 will now fire duplicate events and when I remove the "proxy light dom" the click event is now triggered as expected!